linking error with Utilities::MPI::min only in release mode

The following code compiles and runs perfectly in debug mode
```c++
#include <deal.II/base/mpi.h>

using namespace dealii;

int main(int argc, char ** argv)
{
  Utilities::MPI::MPI_InitFinalize mpi_initialization(
    argc, argv, numbers::invalid_unsigned_int);
  AssertThrow(Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD) == 2, ExcMessage("This program must run with two MPI processes"));
  const auto mpi_rank = Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
  std::vector<unsigned int> v;
  if (mpi_rank == 0)
  {
    v.push_back(1);
    v.push_back(2);
  }
  else
  {
    v.push_back(3);
    v.push_back(4);
  }
  ArrayView<const unsigned int> v_const_view(v);
  ArrayView<unsigned int> v_view(v);
  Utilities::MPI::min<unsigned int>(v_const_view, MPI_COMM_WORLD, v_view);
  for (auto i : v)
    std::cout << i << ", ";
  std::cout << std::endl;
  return 0;
}
```
but fails in release mode with the error:
```
CMakeFiles/arrayview_min.dir/arrayview_min.cc.o: In function `main':
/home/reza/repos/dealsims/dealii-examples/arrayview_min/arrayview_min.cc:24: undefined reference to `void dealii::Utilities::MPI::min<unsigned int>(dealii::ArrayView<unsigned int const, dealii::MemorySpace::Host> const&, ompi_communicator_t* const&, dealii::ArrayView<unsigned int, dealii::MemorySpace::Host> const&)'
collect2: error: ld returned 1 exit status
```
could be related to #2739. I will try to add some function template instantiations.

--- a/source/base/mpi.inst.in
+++ b/source/base/mpi.inst.in
@@ -51,11 +51,19 @@ for (S : MPI_SCALARS)
                                       const MPI_Comm &,
                                       std::vector<S> &);
 
+    template void max<S>(const ArrayView<const S> &,
+                         const MPI_Comm &,
+                         const ArrayView<S> &);
+
     template S min<S>(const S &, const MPI_Comm &);
 
     template void min<std::vector<S>>(const std::vector<S> &,
                                       const MPI_Comm &,
                                       std::vector<S> &);
+
+    template void min<S>(const ArrayView<const S> &,
+                         const MPI_Comm &,
+                         const ArrayView<S> &);
 
     // The fixed-length array (i.e., things declared like T(&values)[N])
     // versions of the functions above live in the header file mpi.h since the

